pipeline/test/test_processor.py

```diff
@@ -0,0 +1,790 @@
+import os
+from pathlib import Path
+
+import pandas as pd
+
+from unittest.mock import patch
+from extract.document import Document, Page, TextBlock
+from processor.postprocessor import (
+    HyphenationPostProcessor,
+    AdobeListGroupingPostProcessor,
+    AdobeTextStylingPostProcessor,
+)
+import pytest
+
+
+@pytest.fixture
+def test_input_path() -> Path:
+    return (
+        Path(__file__).parent / "data/cclw-1318-d7f66920a18e4ddf94c83cf21fa2bcfa.json"
+    )
+
+
+# Keep these as fixtures in case they get more complex in the future.
+@pytest.fixture
+def adobe_list_postprocessor() -> AdobeListGroupingPostProcessor:
+    return AdobeListGroupingPostProcessor()
+
+
+@pytest.fixture
+def adobe_text_styling_postprocessor() -> AdobeTextStylingPostProcessor:
+    return AdobeTextStylingPostProcessor()
+
+
+@pytest.fixture
+def adobe_hyphenation_postprocessor() -> HyphenationPostProcessor:
+    return HyphenationPostProcessor()
+
+
+def test_hyphenation_postprocessor_rewrap_hyphenated_words(document):
```

Comment:
I deliberately chose an example that this processor doesn't handle perfectly to illustrate some of the arbitrariness of the Adobe results (see comments).